libs/natparse/src/main/java/org/amshove/natparse/parsing/StatementListParser.java

```diff
@@ -2143,14 +2146,40 @@ private SetKeyStatementNode setKey() throws ParseError
 		consumeMandatory(statement, SyntaxKind.SET);
 		consumeMandatory(statement, SyntaxKind.KEY);
 
+		if (consumeAnyOptionally(statement, List.of(SyntaxKind.ALL, SyntaxKind.ON, SyntaxKind.OFF)))
+		{
+			return statement;
+		}
 		if (consumeOptionally(statement, SyntaxKind.NAMED))
 		{
 			consumeMandatory(statement, SyntaxKind.OFF);
+			return statement;
+		}
+		if (consumeOptionally(statement, SyntaxKind.COMMAND))
+		{
+			consumeAnyMandatory(statement, List.of(SyntaxKind.ON, SyntaxKind.OFF));
+			return statement;
 		}
 
-		while (peekKind(SyntaxKind.PF))
+		while (peekKind(SyntaxKind.IDENTIFIER) || peekKind(SyntaxKind.DYNAMIC))
```

Comment:
The following raises a false-positive diagnostic for me:

```
SET KEY PF5 NAMED 'PHON'
#HEADER2 := #HEADER2-NORM /* ERROR;Unexpected token #HEADER2, expected one of PFnn, PAn, CLR
```

this should be resolved if you change the condition to:
`while ((peekKind(SyntaxKind.IDENTIFER) && !isStatementStart()) || peekKind(SyntaxKind.DYNAMIC))`
Instead of `isStatementStart()` you can also use `isAssignmentStart()`, which only handles the case `IDENTIFIER COLON_EQUALS`, but it is called by `isStatementStart()` anyway :-)